src/fdcache_entity.cpp

```diff
@@ -2552,6 +2570,8 @@ bool FdEntity::AddUntreated(off_t start, off_t size)
 
 bool FdEntity::GetLastUpdateUntreatedPart(off_t& start, off_t& size) const
 {
+    const std::lock_guard<std::mutex> lock(fdent_lock);
```

Comment:
This method is called as a callback from PseudoFdInfo::UploadBoundaryLastUntreatedArea().
The FdEntity object passed to PseudoFdInfo::UploadBoundaryLastUntreatedArea() has already performed this locking, so it cannot be locked here.